### ISSUE TITLE
Crash with `fgetxattr: No data available` when reusing .flatpak-builder/ from a previous run

### ISSUE THREAD

**Issue by jhasse** [flatpak/flatpak-builder#57]
In order to not rebuild my dependencies all the time, I let CI cache the `.flatpak-builder` directory. This results in the following crash though:

```
Starting build of org.wxformbuilder.wxFormBuilder
Cache hit for wxWidgets, skipping build
]2;flatpak-builder: Updating wxWidgetsCache miss, checking out last cache hit

(flatpak-builder:266): flatpak-builder-ERROR **: Failed to check out cache: fgetxattr: No data available
./flatpak.sh: line 8:   266 Trace/breakpoint trap   (core dumped) flatpak-builder --repo=repo ./build org.wxformbuilder.wxFormBuilder.json
```

See https://travis-ci.org/wxFormBuilder/wxFormBuilder/jobs/296809091

Would be great if flatpak-builder could handle that situation.

**Comment by mwleeds** [comment]
You could try setting `FLATPAK_OSTREE_REPO_MODE=user-only` in the environment so flatpak won't use extended file attributes. It sounds like the CI isn't preserving them correctly.

**Comment by vrutkovs** [comment]
Setting `FLATPAK_OSTREE_REPO_MODE` didn't seem to help here - https://travis-ci.org/vrutkovs/io.atom.electron.BaseApp/builds/297197916 (cache was cleared and rebuilt with this env var set too)

**Comment by jhasse** [comment]
Same for me: https://travis-ci.org/wxFormBuilder/wxFormBuilder/jobs/297199520 after https://github.com/wxFormBuilder/wxFormBuilder/commit/8557be638182d25c4adcf296975cae0698c9123e

**Comment by vrutkovs** [comment]
It seems builder ignores this env var and always creates cache with [`OSTREE_REPO_MODE_BARE_USER`](https://github.com/flatpak/flatpak-builder/blob/master/src/builder-cache.c#L251), while the env var is [used](https://github.com/flatpak/flatpak/blob/master/common/flatpak-dir.c#L1593) to install runtimes/apps

**Comment by vrutkovs** [comment]
Indeed, `flatpak-builder` ignores this env var when cache is opened. `vrutkovs/flatpak:test` docker image has a patched version of the builder and seems to fix the issue for me

**Comment by alexlarsson** [comment]
Using user-only for the cache is not a good idea. It is a lossy mode that creates broken repositories unless you are very careful. It is likely to cause broken builds.

If the CI doesn't correctly persist the xattrs i would recommend manually persisting them in a way that does. For instance, tar up the cache at the end.

**Comment by vrutkovs** [comment]
Even if the cache is tarred its being extracted on Travis CI VM's filesystem, which has attrs disabled

**Comment by jhasse** [comment]
Hm ... works for me though (note how wxWidgets isn't build again): https://travis-ci.org/wxFormBuilder/wxFormBuilder/jobs/297872450
This is how I did it: https://github.com/wxFormBuilder/wxFormBuilder/commit/11f4c8c246a20af6827f5818faee4535420ec8ef
